### ISSUE TITLE
dd-trace interferes with AsyncLocalStorage in Hapi.js Applications

### ISSUE THREAD

**Issue by cristunaranjo** [DataDog/dd-trace-js#4557]
Using `dd-trace` in a `Hapi.js` application causes issues with `AsyncLocalStorage` (ALS). Specifically, `ALS` contexts do not persist across asynchronous boundaries, resulting in undefined or incorrect values when attempting to retrieve stored data. This behaviour disrupts applications relying on `ALS` for request-specific context management.

**Steps to Reproduce:**

- Set up a `Hapi.js` server.
- Initialize `AsyncLocalStorage` to store request-specific data.
- Integrate `dd-trace` for tracing.
- Attempt to access ALS-stored data within route handlers.

**Example Code:**

```js
require("dd-trace").init();

const Hapi = require("@hapi/hapi");
const { AsyncLocalStorage } = require("async_hooks");

const als = new AsyncLocalStorage();

const init = async () => {
  const server = Hapi.server({
    port: 3000,
    host: "localhost",
  });

  server.ext("onRequest", (request, h) => {
    als.enterWith({ path: request.path });
    return h.continue;
  });

  server.route({
    method: "GET",
    path: "/",
    handler: async (request, reply) => {
      const store = als.getStore();
      console.log("store", store);
      if (!store) {
        throw new Error("ALS store is undefined");
      }

      return store;
    },
  });

  await server.start();
  console.log("Server running on %s", server.info.uri);
};

process.on("unhandledRejection", (err) => {
  console.log(err);
  process.exit(1);
});

init().catch(console.error);
```

**Expected Behavior:**
The `AsyncLocalStorage` store should persist and be accessible within the route handler, maintaining context for each request.

**Actual Behavior:**
The `AsyncLocalStorage` store is undefined within the route handler when `dd-trace` is enabled.

**Impact:**
This issue prevents the use of `AsyncLocalStorage` for request-specific context management in `Hapi.js` applications instrumented with `dd-trace`, disrupting functionalities such as logging, tracing, and request-specific data storage.

**Environment:**

Node.js version: 18.20.3
dd-trace version: 5.19.0
Hapi.js version: 21.3.10
Operating System: macOS Sonoma (Version 14.5)

**Additional Context:**
This issue seems similar to previously reported issues with `AsyncLocalStorage` in other frameworks and libraries, as seen in [issue #2056](https://github.com/DataDog/dd-trace-js/issues/2056). It appears that `dd-trace` may be interfering with the asynchronous context propagation that AsyncLocalStorage relies on.

**Steps Taken to Troubleshoot:**

- Verified `ALS` works without `dd-trace`.
- Reproduced issue consistently with `dd-trace` enabled.
- Followed various online resources and recommendations without success.

**Comment by simon-id** [comment]
@cristunaranjo I believe the fix you proposed was merged and release. Can you confirm this issue is closed ?

**Comment by cristunaranjo** [comment]
Yes, it was merged here https://github.com/DataDog/dd-trace-js/commit/a04ba3f9e7fbf4a38f20aef997c8b6d5f50ed7a1

We have being using it now for a while with no issues. Thanks!
